### DISCUSSION TITLE
Image overlay with camera

### DISCUSSION THREAD

**Discussion by ThSGM** [tomshaffner/PiThermalCam#23] [Ideas]
Thank you to @tomshaffner for writing such an interesting guide for the camera, including the discussions about the home insulation. 

I am wondering if there was any scope for a feature of essentially either:

1. Overlaying the thermal images on a live camera feed
2. Incorporating the thermal camera edge detection feature implemented here: https://github.com/obstruse/ThermalCamera into the PiThermalCam software

It seems to me that the big problems are twofold:

1. It's very hard to recognize what the MLX90640 sensor is pointing at -- the edge detection feature deals with that, allowing you to see the image behind
2. The low fps of the sensor

Moreover, it seems to me that in most applications, like looking for insulation cold spots, we don't need high fps. 

So I wonder if, instead of having to calculate edge detection, we would be better off simply overlaying the thermal colours onto a high fps camera feed. So you have your thermal colours refreshed every 1-4 fps say, while your webcam or picam is running at a faster fps.

**Comment by tomshaffner** [comment]
@ThSGM, I love this idea but the project itself was generally targeted at the cheapest way to get to a thermal camera image. Mounting a second camera and then aligning it with the first exactly would be great but is a fair bit of an expansion on what there is now. If I get back up this project I'll keep that in mind as an idea for next, and in the meantime it's open source so you're welcome to make the changes yourself and submit a pull request!

In the meantime, my trick to find edges was simply to point the camera at something known (e.g. a door) and then, once I'd found that thing in the image, move the camera slowly enough to be able to track what I'm looking at. That and some strategic moving closer and further away was generally sufficient for me to identify everything in the image. Much less elegant than what you're suggesting above, but sufficient for my purposes. My real hope was that the next generation of these cameras with much more resolution would come out soon and I could fairly easily swap out for a better one.

**Reply by tomshaffner** [reply]
@ThSGM, the edge detection point is a very cool one. More so, it seems open cv has the algorithm for it so in one of the current modes it could be added in fairly easily. Think I'll open a new issue for this. Not sure if/when I'd get back to it, but when I do, or if anyone else does, it'll be queued up. Thanks for the suggestion!!
